homeassistant/components/rfxtrx/__init__.py

```diff
@@ -140,7 +141,8 @@ def _ensure_device(value):
 )
 
 CONFIG_SCHEMA = vol.Schema(
-    {DOMAIN: vol.Any(DEVICE_SCHEMA, PORT_SCHEMA)}, extra=vol.ALLOW_EXTRA
+    {DOMAIN: vol.All(cv.deprecated(CONF_DEBUG), vol.Any(DEVICE_SCHEMA, PORT_SCHEMA))},
```

Comment:
@MartinHjelmare I can't wrap my head around it...any clue what goes wrong here? I both get a deprecated warning and an error.
```
2020-09-28 18:51:27 WARNING (MainThread) [homeassistant.components.rfxtrx] The 'debug' option is deprecated, please remove it from your configuration
2020-09-28 18:51:27 ERROR (MainThread) [homeassistant.config] Invalid config for [rfxtrx]: [debug] is an invalid option for [rfxtrx]. Check: rfxtrx->rfxtrx->debug. (See /workspaces/core/config/configuration.yaml, line 17). Please check the docs at https://www.home-assistant.io/integrations/rfxtrx
2020-09-28 18:51:27 ERROR (MainThread) [homeassistant.setup] Setup failed for rfxtrx: Invalid config.
```

Reply:
It's ugly. But you should do it on base schema

```
BASE_SCHEMA = vol.All(
  vol.deprecated(CONF_DEBUG),
  {
        vol.Ignore(CONF_DEBUG): cv.boolean,
        vol.Optional(CONF_AUTOMATIC_ADD, default=False): cv.boolean,
        vol.Optional(CONF_DEVICES, default={}): {cv.string: _ensure_device},
  }
)
```

It would be nice if that deprecation validation thing could be used easier :)

Reply:
`vol.All` does not work here because the `BASE_SCHEMA` is extended. Also `vol.Ignore` does not seem to exist. I can get it working by keeping it `vol.Optional` and add it as `cv.deprecated` as above. Pushing now...